### ISSUE TITLE
VB.Net "Hide Advanced Members" is always on

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#40497]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/854579/vbnet-hide-advanced-members-is-always-on.html)._

---
Under Settings - Text editor - Basic - General, there is a setting, &quot;Hide advanced members&quot;.

This setting is disabled (greyed out) and unckeched.
Yet, the VS behaves as if it is Enabled. That is, &quot;advanced&quot; members of all classes are hidden from VB.NET users in all projects. Intellisense does not show them, and there is no way to know they even exist.

I've tried exporting the VS settings. In the exported file, the setting is listed as &quot;false&quot; (which matches the unchecked look of the textbox).

I've tried resetting the settings back to defaults. That did not fix the behaviour, IntelliSense still incorrectly hides &quot;advanced&quot; members in VB projects.

This makes programming in VB.NET (mostly supporting existing codebase) really, really frustrating.

Expected behaviour:
All members are shown in VB.NET, or at least there is a way to make them show. Currently there is no way to make them show.

A fine example of a hidden &quot;advanced&quot; member is DataGridViewRow.DataBoundItem (in a Windows Forms application):

```VB
Dim r As DataGridViewRow
r.Da   ' Intellisense appears, but DataBoundItem is not listed.
```

I am observing the same problem in VS 2019.

Like in 2017, there is simply no way to have Intellisence show the members there.

---
#### Visual Studio Feedback System on 12/13/2019, 02:11 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 12/18/2019, 03:17 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

---
### Original Solutions
(no solutions)

**Comment by dpoeschl** [comment]
I can't remember if there was some old reason why we went with this behavior, but it does seem incorrect.

**Comment by slepmog** [comment]
> I can't remember if there was some old reason why we went with this behavior, but it does seem incorrect.

I will *speculate* that you wanted to hide pointer-related stuff from VB users having no plans to support it there in the foreseeable future, but forgot that the "advanced member" flag is widely used in the framework for things not related to pointers.

**Comment by AliParLouS** [comment]
A news about this issue? It almost 2 years visual studio have this problem and still does in 16.5.4.

**Comment by AliParLouS** [comment]
If anyone had this issue in VB, Close VS 2019, Open this file with notepad
`C:\Users\%username%\AppData\Local\Microsoft\VisualStudio\16.0_1442dd31\Settings\CurrentSettings.vssettings`
 and search for 
`name="HideAdvancedMembers">false`
 then change false to true, Save the file, Then all intellisense should work in VB.

The version 16.0_1442dd31 could be different in you PC, You should find right version.

**Comment by phildaboy** [comment]
@AliParLouS I opened that file in both Visual Studio and Notepad++ and it was already true so I changed it to false which didn't work then changed it back to true and it still didn't work. And when I tried to open the file in Notepad there wasn't anything in C:\Users\%username%\AppData\Local\Microsoft\VisualStudio\16.0_1442dd31\Settings.

**Comment by tiggronline** [comment]
Still an issue in VS2019 v16.7.7

**Comment by nikoboychev** [comment]
Hello everyone. I just start to learn coding and try to use VS and i have same error like you. Dont have suggestions for coding. I try to do everything that i can find to fix it, but it doesnt work. And finaly when i see there is no answer for 2 years i decide to change the editor. Something strange happened. I try another Microsoft product "Visual Studio Code". When i install it he ask me did i want to install some extension to write in C#. Ofcourse i click yes and after that i dont know why i decide to try Visual Studio again and it worked just fine with all code suggestions. Hope this work for you too or I give some clue how can this be fixed.

**Comment by jimmycw74** [comment]
Recently I had to install VS in a new laptop and this problem came up. I can't change the property "Hidden advanced members" under certain languages and if I change it in "All languages" nothing is fixed. Main language of the problematic project is VB.net.
I changed directly in the property file as suggested by AliParLouS  but also doesn't work.

My current setup :
Microsoft Visual Studio Community 2019 Versão 16.10.1
VisualStudio.16.Release/16.10.1+31402.337 
Microsoft .NET Framework Versão 4.8.04084
Versão Instalada: Community
ASP.NET and Web Tools 2019   16.10.526.50910
ASP.NET and Web Tools 2019
ASP.NET Web Frameworks and Tools 2019   16.10.526.50910
Depuração do .NET Core com WSL 2   1.0
Extensão do IntelliCode   1.0
Ferramentas do TypeScript   16.0.30429.2002
Ferramentas do Visual Basic   3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb
